### ISSUE TITLE
Feature Request - Support a Wider Range of Hash Policies

### ISSUE THREAD

**Issue by charlieoleary** [datawire/ambassador#422]
Currently, with the v1 Envoy API, `hash_policy` annotations are limited to `header`. The v2 API offers additional hash policies including `cookie` and `source_ip`.  Additionally, both with the v1 and v2 APIs an `envoy_override` is required to actually implement this. Native support in annotations would be ideal.

[Envoy V1 API hash_policy](https://www.envoyproxy.io/docs/envoy/latest/api-v1/route_config/route#config-http-conn-man-route-table-hash-policy)
[Envoy v2 API HashPolicy](https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/route/route.proto#route-routeaction-hashpolicy)

**Comment by richarddli** [comment]
Thanks for the feature request! We are working on a switch to the v2 APIs / config format at the moment, so we'd prefer to wait until the switch is done.

In the meantime, have you been able to get this to work with `envoy_override`? I'd be happy to add a documentation example if this is something you're using.

**Comment by charlieoleary** [comment]
@richarddli That's good to know and I'm excited to hear you are working on it! I tried a few different approaches with `envoy_override` but didn't have any luck. Is it possible to use the overrides with the v2 API commands? If so, I'd definitely love to see an example.

**Comment by richarddli** [comment]
@charlieoleary unfortunately, not at the moment. are you able to use Ambassador at the moment without this feature? we're trying to prioritize the work on supporting the v2 APIs, so any context is really helpful.

**Comment by charlieoleary** [comment]
Yeah I figured that was the case. Unfortunately a portion of our app does require sticky sessions, so implementing this is a blocker for us until that’s added. IP hash won’t work, which is the only type of hashing that the v1 API supports.

**Comment by kflynn** [comment]
(Ignore that unintentional close!) Here's an actual V1 `hash_policy` used by @adaniline (cf #951):

```
        envoy_override:
          hash_policy:
            cookie:
              name: "NODE_ID"
              ttl: 
                seconds: 3600
              path: "/myapp/socket.io"
```

We should support `hash_policy` without `envoy_override` for 0.50.0.

**Comment by kflynn** [comment]
Dropping from the `0.50.0 GA` milestone; we'll add this to a release shortly after GA.

**Comment by Tapppi** [comment]
@kflynn any updates regarding this? You say that `envoy_override.hash_policy` should be supported by 0.50 in #951 , but there are no docs for session affinity at getambassador.io 😞 Looking at envoy v2 configs, they [support affinity well](https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/route/route.proto#route-routeaction-hashpolicy) but I have no idea how I would use that in ambassador.

**Comment by olavim** [comment]
@kflynn Are there any up-to-date docs on what, if any, `envoy_override`s are supported currently?

**Comment by fraywing** [comment]
I'm waiting for this as well, stoked to see it get in (or native ambassador session affinity)

**Comment by richarddli** [comment]
This is now available as of 0.52.
